### DISCUSSION TITLE
Android 10 location permission: "Allow All the Time"

### DISCUSSION THREAD

**Discussion by jrzelazny** [osmandapp/OsmAnd#19146] [General]
I'm using an older Android phone using Android 10.

It seems that at some point the option for OsmAnd to use the location "All the time" has gone away. ]
<table style="width:100%">
<tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/146493055/36f877fe-f04c-4bc5-93d2-2d68f6166ba1"  width="300" height="auto"/> </th>
  </tr>
</table>

So it means every time I go to another app, even if briefly, the app loses the location and when I come back it takes quite a bit of time before it puts me back in the right location.

Or maybe it's a problem with using the app in the background, as my newer Android device also doesn't have "All the time"  but it works fine. It's only on the older device that the location is lost every time I open up a different app.

Any suggestions?

**Comment by DmitryAlexei** [comment]
Please disable battery optimization for OsmAnd in Android settings.

**Reply by jrzelazny** [reply]
Yes, all of that has been turned off. I've turned off every optimization and battery saver mode I can find. It's made no difference.

I have several navigation apps installed on the phone, and a couple of the apps suffer from this problem, and a couple don't. So I don't think this is a OsmAnd-specific problem, (un)fortunately. 

I was hoping I could find a way to force OsmAnd to Always Use Location... but I can't find any way to do so. And being an older version of Android, there's not a lot of new information I can find how to do this.

**Reply by DmitryAlexei** [reply]
Please try to change Location source from Google Play Services to Android API (Menu > Settings > OsmAnd settings > Location services). Also update your OsmAnd to the actual version from Google play (you can also try beta).

**Reply by jrzelazny** [reply]
Thanks! I'm pretty sure in all my attempts to get it to work I tried switching to Android API... though maybe I didn't. There are so many different options and I tried different settings on my phone and in the app.

I'll switch to Android API and give it a test and report back.

I'm not quite sure what you mean by updating my app to the actual version. I downloaded the app from the Google Play store, and it's on the latest version. Is that what you mean?

**Reply by jrzelazny** [reply]
Wow, thank you. That fixed my problem. So simple!

I had made numerous attempts to get the app to function, and nothing seemed to help, but your suggestion solved it!
